### DISCUSSION TITLE
Welcome to bl_ui_widgets Discussions!

### DISCUSSION THREAD

**Discussion by mmmrqs** [mmmrqs/bl_ui_widgets#1] [Announcements]
Hi there, Welcome to my GitHub projects. Feel free to create issues, file requests etc. I hope you enjoy it and that it may be helpful to your own projects.

**Comment by Oxicid** [comment]
Hi, I decided to create a small but very important interface addon for modular house building, but one problem remains. I have buttons with necessary snapping value, and I have slider with certain value range, slider works fine, but when I want to click on buttons and then set values from slider without changing its value (i.e. without touching necessary value), I can not pass its value with "Set" button. I am very bad at programming and I would like to know what lines of code I need to write to make the Set button work? This addon is based on Jayanam code, although I like your version better and it does not lag, but from Jayanam everything looks easie
[Grid_Scale_Switcher.zip](https://github.com/mmmrqs/bl_ui_widgets/files/8037951/Grid_Scale_Switcher.zip)
r

**Reply by mmmrqs** [reply]
Hi Oxicid.  Thanks for reaching out to me, and for your interest in my little addon.  I am not sure I've understood exactly what is your need, so please describe the process in steps.  For those buttons which you mentioned that they have snapping values, are those fixed values (example one button has a fixed value of 100 and the other button has a fixed value of 150) so when you press a given button it would cause the slider to automatically be set to that fixed value ?  Or do you want an entry field where the user would type a value and then press a generic "Set" button which would finally cause the slider to be set to that entered value ?  Or is it the other way around where you actually want the value from the slider to be captured by the "Set" button and thus to have that value set on something else (what would be the destination for the "Set" value set by the button?).  Please help me understand the flow.  You may want to describe it as in user-taken-steps, please.  Appreciated it.

**Reply by Oxicid** [reply]
Need the current value from the slider when you click on Set. The whole point is that I want to set the slider to the required value, and periodically switch between the already preset buttons, without touching the slider (just on the slider the user will set the required value, and it is already again do not want to pull after setting) I'm somewhere close, but I don't know what to do next :)    

def button6_press(self, widget, slider):
        currents_sld_value = BL_UI_Slider()
        sld_set = currents_sld_value.get_value
        bpy.context.space_data.overlay.grid_scale = sld_set

**Reply by mmmrqs** [reply]
Got it.  You just need a little change to the code. I have tried it and it worked now as you expected.
On your "drag_panel_op.py" file do as follows:

    def button6_press(self, widget):
        bpy.context.space_data.overlay.grid_scale = self.slider.get_value()
        
The "slider" object is defined at the top so it is visible to all other functions inside drag_paneLop.py.  You can get or set its value by making a reference to it and calling the appropriate slider's function, in this case it was the "get_value()".  You can find all other available functions by looking at the code of "bl_ui_slider.py".  Hope this helps, nice addon you've got there.

**Comment by Oxicid** [comment]
You have no idea how much you have made me happy, I learned so much in such a short period of time, thank you that there are people like you :). Everything works as it should, and I was almost close.

**Reply by mmmrqs** [reply]
Cool.  I'm glad it was helpful.  And yes I do have a pretty good idea about the feeling because I felt the same when I found out Jayanam's addon. It was exactly what I needed for the project I've been modelling in Blender for months.  Without it I don't think I would be able to complete my mission LOL. That's the way. One people helps another which helps the next one and so on, like a good chain reaction.
